<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML
><HEAD
><TITLE
>NCBI BLAST</TITLE
><META
NAME="GENERATOR"
CONTENT="Modular DocBook HTML Stylesheet Version 1.79"><LINK
REL="HOME"
TITLE=" Bio Users Guide "
HREF="index.html"><LINK
REL="UP"
TITLE="Using"
HREF="using.html"><LINK
REL="PREVIOUS"
TITLE="HMMER"
HREF="hmmer_usage.html"><LINK
REL="NEXT"
TITLE="ClustalW"
HREF="clustalw_usage.html"><LINK
REL="STYLESHEET"
TYPE="text/css"
HREF="rocks.css"></HEAD
><BODY
CLASS="SECTION"
BGCOLOR="#FFFFFF"
TEXT="#000000"
LINK="#0000FF"
VLINK="#840084"
ALINK="#0000FF"
><DIV
CLASS="NAVHEADER"
><TABLE
SUMMARY="Header navigation table"
WIDTH="100%"
BORDER="0"
CELLPADDING="0"
CELLSPACING="0"
><TR
><TH
COLSPAN="3"
ALIGN="center"
><B
CLASS="COMMAND"
>Bio</B
> Users Guide: 	<DIV
CLASS="MEDIAOBJECT"
><P
><IMG
SRC="images/dna.png"></P
></DIV
>
	</TH
></TR
><TR
><TD
WIDTH="10%"
ALIGN="left"
VALIGN="bottom"
><A
HREF="hmmer_usage.html"
ACCESSKEY="P"
>Prev</A
></TD
><TD
WIDTH="80%"
ALIGN="center"
VALIGN="bottom"
>Chapter 3. Using</TD
><TD
WIDTH="10%"
ALIGN="right"
VALIGN="bottom"
><A
HREF="clustalw_usage.html"
ACCESSKEY="N"
>Next</A
></TD
></TR
></TABLE
><HR
ALIGN="LEFT"
WIDTH="100%"></DIV
><DIV
CLASS="SECTION"
><H1
CLASS="SECTION"
><A
NAME="BLAST_USAGE"
>3.3. NCBI BLAST</A
></H1
><DIV
CLASS="SECTION"
><H2
CLASS="SECTION"
><A
NAME="AEN169"
>3.3.1. About</A
></H2
><P
>BLAST, or Basic Local Alignment Search Tool, is a collection of
tools that are used to search for and find
regions of local similarity between sequences. The program
compares nucleotide or protein sequences to sequence  databases,
and calculates the statistical significance of the matches. This 
software suite has been released free to the public by the
National Centre for Biotechnology Information.</P
></DIV
><DIV
CLASS="SECTION"
><H2
CLASS="SECTION"
><A
NAME="AEN172"
>3.3.2. Usage</A
></H2
><P
>BLAST can be used for protein-protein comparisons or
nucleotide-nucleotide comparisons. Before an example of the usage
is presented, we must first define some environmental variables.</P
><P
></P
><UL
><LI
><P
>		$BLASTDB - This is the variable which points to
		the Blast Database. This is set to
		$HOME/bio/ncbi/db/. This directory should contain
		the databases that you would want to search. BLAST,
		by default, checks this location and the current working
		directory for the
		presence of the databases. This variable is set during
		login by system login scripts , and may be changed
		by the user to point to her preferred location in
		her startup scripts.
	</P
></LI
><LI
><P
>		$BLASTMAT - This variable points to the location
		where the BLAST scoring matricies are present. It is set to
		/opt/bio/ncbi/data.
		Again, they may be changed to point to a desired
		location on a per-user basis.
	</P
></LI
></UL
><P
>	BLAST requires the presence of 2 datasets. One dataset is
	the input sequence that you want to search for, and the other
	dataset is the database that you want to search against.
	</P
><P
> Use the following procedure to run blast
	<P
></P
><UL
><LI
><P
>Download a BLAST database that
	you want to run the comparison against.
	The databases can be obtained from the NCBI ftp site at 
	<A
HREF="ftp://ftp.ncbi.nlm.nih.gov/blast/db/"
TARGET="_top"
>ftp://ftp.ncbi.nlm.nih.gov/blast/db/</A
>.</P
><DIV
CLASS="NOTE"
><P
></P
><TABLE
CLASS="NOTE"
WIDTH="90%"
BORDER="0"
><TR
><TD
WIDTH="25"
ALIGN="CENTER"
VALIGN="TOP"
><IMG
SRC="./stylesheet-images/note.png"
HSPACE="5"
ALT="Note"></TD
><TD
ALIGN="LEFT"
VALIGN="TOP"
><P
>The databases available on the site mentioned above are pre-formatted.</P
><P
>It is recommended that the blast databases be
	stored at the $BLASTDB location.</P
></TD
></TR
></TABLE
></DIV
><P
>Visit <A
HREF="ftp://ftp.ncbi.nlm.nih.gov/blast/db/"
TARGET="_top"
>ftp://ftp.ncbi.nlm.nih.gov/blast/db/</A
>
	in your browser to
	see a list of available preformatted databases.</P
><P
>Download
	one of these on to your cluster using wget.</P
><TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>[nostromo@xxx ~]$ wget -q ftp://ftp.ncbi.nlm.nih.gov/blast/db/nt.08.tar.gz
[nostromo@xxx ~]$ gunzip -c nt.08.tar.gz | ( cd $BLASTDB/ &#38;&#38; tar -xf -)</PRE
></FONT
></TD
></TR
></TABLE
></LI
><LI
><P
>The above method downloads a formatted database, and untars
	it into $BLASTDB.</P
><P
>Unformatted databases can be obtained in FASTA format at
	<A
HREF="ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/"
TARGET="_top"
>	ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/</A
>.
	</P
><P
>Visit <A
HREF="ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/"
TARGET="_top"
>	ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/</A
> in your web browser</P
><DIV
CLASS="NOTE"
><P
></P
><TABLE
CLASS="NOTE"
WIDTH="90%"
BORDER="0"
><TR
><TD
WIDTH="25"
ALIGN="CENTER"
VALIGN="TOP"
><IMG
SRC="./stylesheet-images/note.png"
HSPACE="5"
ALT="Note"></TD
><TD
ALIGN="LEFT"
VALIGN="TOP"
><P
>If you've downloaded the databases from 
	ftp://ftp.ncbi.nlm.nih.gov/blast/db/, then DO NOT run formatdb.</P
></TD
></TR
></TABLE
></DIV
><P
>Run the formatdb
	command to format the database to the BLAST format. For this example,
	we'll use the Drosophila Melanogaster (fruitfly) nucleotide database</P
><TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>[nostromo@xxx ~]$ cd $BLASTDB
[nostromo@xxx ~]$ wget -q ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA/drosoph.nt.gz
[nostromo@xxx ~]$ gunzip drosoph.nt.gz 
[nostromo@xxx ~]$ formatdb -p F -V T -i drosoph.nt 
[nostromo@xxx ~]$ ls drosoph.nt*
drosoph.nt  drosoph.nt.nhr  drosoph.nt.nin  drosoph.nt.nsq
[nostromo@xxx ~]$ cd $HOME</PRE
></FONT
></TD
></TR
></TABLE
></LI
><LI
><P
>After the database is formatted, create a test input file.</P
><TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>[nostromo@xxx ~]$ cat &#62; test.txt
&#62;Test
AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGC
TTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTAGGTCACTAAATACTTTAACCAA
TATAGGCATAGCGCACAGACAGATAAAAATTACAGAGTACACAACATCCATGAAACGCATTAGCACCACC
ATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGACGCGTACAGGAAACACAGAAAAAAG
CCCGCACCTGACAGTGCGGGCTTTTTTTTTCGACCAAAGGTAACGAGGTAACAACCATGCGAGTGTTGAA
GTTCGGCGGTACATCAGTGGCAAATGCAGAACGTTTTCTGCGTGTTGCCGATATTCTGGAAAGCAATGCC
AGGCAGGGGCAGGTGGCCACCGTCCTCTCTGCCCCCGCCAAAATCACCAACCACCTGGTGGCGATGATTG
AAAAAACCATTAGCGGCCAGGATGCTTTACCCAATATCAGCGATGCCGAACGTATTTTTGCCGAACTTTT
	</PRE
></FONT
></TD
></TR
></TABLE
></LI
><LI
><P
>Run the blastall program on the test input against
	the formatted database.</P
><TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>[nostromo@xxx ~]$ blastall --help
	</PRE
></FONT
></TD
></TR
></TABLE
><P
>	gives a list of all the options that you can use to run
	the blastall program.</P
><TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>[nostromo@xxx ~]$  blastall -d drosoph.nt -p blastn -i test.txt 
BLASTN 2.2.18 [Mar-02-2008]


Reference: Altschul, Stephen F., Thomas L. Madden, Alejandro A. Schaffer, 
Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman (1997), 
"Gapped BLAST and PSI-BLAST: a new generation of protein database search
programs",  Nucleic Acids Res. 25:3389-3402.

Query= Test
         (560 letters)

Database: drosoph.nt 
           1170 sequences; 122,655,632 total letters

Searching..................................................done



                                                                 Score    E
Sequences producing significant alignments:                      (bits) Value

gi|10729531|gb|AE002936.2|AE002936 Drosophila melanogaster genom...    36   0.86 
gi|10728232|gb|AE003493.2|AE003493 Drosophila melanogaster genom...    36   0.86 
gi|10726497|gb|AE003698.2|AE003698 Drosophila melanogaster genom...    36   0.86 
gi|10726398|gb|AE003681.2|AE003681 Drosophila melanogaster genom...    36   0.86 
gi|10729308|gb|AE002665.2|AE002665 Drosophila melanogaster genom...    34   3.4  
gi|10729264|gb|AE002615.2|AE002615 Drosophila melanogaster genom...    34   3.4  
gi|7298233|gb|AE003648.1|AE003648 Drosophila melanogaster genomi...    34   3.4  
gi|7297628|gb|AE003628.1|AE003628 Drosophila melanogaster genomi...    34   3.4  
gi|10728546|gb|AE003447.2|AE003447 Drosophila melanogaster genom...    34   3.4  
gi|7290819|gb|AE003441.1|AE003441 Drosophila melanogaster genomi...    34   3.4  
gi|10728461|gb|AE003431.2|AE003431 Drosophila melanogaster genom...    34   3.4  
gi|10728241|gb|AE003495.2|AE003495 Drosophila melanogaster genom...    34   3.4  
gi|7292554|gb|AE003484.1|AE003484 Drosophila melanogaster genomi...    34   3.4  
gi|10727872|gb|AE003525.2|AE003525 Drosophila melanogaster genom...    34   3.4  
gi|10727399|gb|AE003587.2|AE003587 Drosophila melanogaster genom...    34   3.4  
gi|10727114|gb|AE003673.2|AE003673 Drosophila melanogaster genom...    34   3.4  
gi|10726705|gb|AE003740.2|AE003740 Drosophila melanogaster genom...    34   3.4  

	</PRE
></FONT
></TD
></TR
></TABLE
><P
>The above example shows how to search for the test input in a drosophila nucleotide database, and a snippet of the output file.
	</P
></LI
></UL
>
	</P
></DIV
><DIV
CLASS="SECTION"
><H2
CLASS="SECTION"
><A
NAME="AEN212"
>3.3.3. Running Blast with SGE</A
></H2
><P
>	This section gives a very simple example of running BLAST through the 
	provided batch system SGE. </P
><P
></P
><UL
><LI
><P
>Create a simple submission script called 
	blast_sge.sh containing the following - 
	<TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>#!/bin/bash
#
#$ -cwd
#$ -S /bin/bash
#$ -j y

export BLASTDB=$HOME/bio/ncbi/db/
export BLASTMAT=/opt/bio/ncbi/data/

/opt/bio/ncbi/bin/blastall -d drosoph.nt \
        -p blastn -i $HOME/test.txt      \
        -o $HOME/result.txt
	</PRE
></FONT
></TD
></TR
></TABLE
></P
></LI
><LI
><P
>	Run
<TABLE
BORDER="0"
BGCOLOR="#E0E0E0"
WIDTH="90%"
><TR
><TD
><FONT
COLOR="#000000"
><PRE
CLASS="SCREEN"
>[nostromo@xxx ~]$ qsub blast_sge.sh 
Your job 10 ("blast_sge.sh") has been submitted</PRE
></FONT
></TD
></TR
></TABLE
></P
></LI
><LI
><P
>The output of the Blast job is similar to the one given above and 
will be stored in $HOME/result.txt</P
></LI
></UL
></DIV
><DIV
CLASS="SECTION"
><H2
CLASS="SECTION"
><A
NAME="AEN224"
>3.3.4. Further Information</A
></H2
><P
>For further information about BLAST and its usage, please refer
	to the following sources
	<P
></P
><UL
><LI
><P
>THE NCBI Blast website - <A
HREF="http://www.ncbi.nlm.nih.gov/BLAST/"
TARGET="_top"
>	http://www.ncbi.nlm.nih.gov/BLAST/</A
></P
></LI
><LI
><P
>BLAST Help page on your cluster <A
HREF="/blast/docs/"
TARGET="_top"
>BLAST Help Page</A
>
	</P
></LI
><LI
><P
>BLAST Program selection Guide - <A
HREF="http://www.ncbi.nlm.nih.gov/blast/BLAST_guide.pdf"
TARGET="_top"
>	http://www.ncbi.nlm.nih.gov/blast/BLAST_guide.pdf</A
></P
></LI
></UL
>
	</P
></DIV
></DIV
><DIV
CLASS="NAVFOOTER"
><HR
ALIGN="LEFT"
WIDTH="100%"><TABLE
SUMMARY="Footer navigation table"
WIDTH="100%"
BORDER="0"
CELLPADDING="0"
CELLSPACING="0"
><TR
><TD
WIDTH="33%"
ALIGN="left"
VALIGN="top"
><A
HREF="hmmer_usage.html"
ACCESSKEY="P"
>Prev</A
></TD
><TD
WIDTH="34%"
ALIGN="center"
VALIGN="top"
><A
HREF="index.html"
ACCESSKEY="H"
>Home</A
></TD
><TD
WIDTH="33%"
ALIGN="right"
VALIGN="top"
><A
HREF="clustalw_usage.html"
ACCESSKEY="N"
>Next</A
></TD
></TR
><TR
><TD
WIDTH="33%"
ALIGN="left"
VALIGN="top"
>HMMER</TD
><TD
WIDTH="34%"
ALIGN="center"
VALIGN="top"
><A
HREF="using.html"
ACCESSKEY="U"
>Up</A
></TD
><TD
WIDTH="33%"
ALIGN="right"
VALIGN="top"
>ClustalW</TD
></TR
></TABLE
></DIV
></BODY
></HTML
>